### ISSUE TITLE
<vaultedFile> does not have a valid extension: yaml, yml, json

### ISSUE THREAD

**Issue by jpswade** [ansible/ansible#18767]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
include_vars_dir

##### ANSIBLE VERSION
```
ansible 2.2.0.0
```

##### CONFIGURATION
default

##### OS / ENVIRONMENT
N/A

##### SUMMARY
The latest version of ansible complains if a vaulted file does not have a valid extension: yaml, yml, json.

The expectation is that a vaulted file may not carry these extensions.

See: https://github.com/ansible/ansible/issues/18223

##### STEPS TO REPRODUCE
```
1. Create a vaulted file that does not carry these extensions.
2. Use include_vars_dir to include it.
3. See error.
```

##### EXPECTED RESULTS
No errors.

##### ACTUAL RESULTS
<vaultedFile> does not have a valid extension: yaml, yml, json

**Comment by alikins** [comment]
This seems to work in stable-2.2 (to be 2.2.1 )and devel branches.

stable-2.2
```
[fedora-23:ansible (stable-2.2 % u=)]$ cat vars_to_include.foo 
blip: 37
baz: 'foobar'
[fedora-23:ansible (stable-2.2 % u=)]$ cat include_var_ping.yml 
---
- hosts: localhost
  gather_facts: false
  tasks:
    - name: include an encrypted vars file with no .yml
      include_vars:
          file: vars_to_include.foo
          name: test_var

    - name: debug unsafe
      debug: var=test_var

[fedora-23:ansible (stable-2.2 % u=)]$ ansible-vault encrypt vars_to_include.foo 
Vault password: 
Encryption successful
[fedora-23:ansible (stable-2.2 % u=)]$ cat vars_to_include.foo 
$ANSIBLE_VAULT;1.1;AES256
33626436376435303330313832653161396461336138353939633466366664346664353865376538
6131326164333032343033353330383933343936353863370a613438346362353733656261626266
64343639396134313039336331376137646632633136333937376563666366636665343333613938
6162643231643131350a356463636662613830636237663066313331346330353330326138633632
61373063306662333733663433373430613266626466383632323363353536326163
[fedora-23:ansible (stable-2.2 % u=)]$ ansible-playbook -vvv include_var_ping.yml --ask-vault-pass
Using /home/adrian/ansible/ansible.cfg as config file
Vault password: 
 [WARNING]: provided hosts list is empty, only localhost is available


PLAYBOOK: include_var_ping.yml *************************************************
1 plays in include_var_ping.yml

PLAY [localhost] ***************************************************************

TASK [include an encrypted vars file with no .yml] *****************************
task path: /home/adrian/src/ansible/include_var_ping.yml:5
ok: [localhost] => {
    "ansible_facts": {
        "test_var": {
            "baz": "foobar", 
            "blip": 37
        }
    }, 
    "changed": false, 
    "invocation": {
        "module_args": {
            "file": "vars_to_include.foo", 
            "name": "test_var"
        }, 
        "module_name": "include_vars"
    }
}

TASK [debug unsafe] ************************************************************
task path: /home/adrian/src/ansible/include_var_ping.yml:10
ok: [localhost] => {
    "test_var": {
        "baz": "foobar", 
        "blip": 37
    }
}

PLAY RECAP *********************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0   

```

devel
```
[fedora-23:ansible (devel % u=)]$ cat vars_to_include.foo 
$ANSIBLE_VAULT;1.1;AES256
33626436376435303330313832653161396461336138353939633466366664346664353865376538
6131326164333032343033353330383933343936353863370a613438346362353733656261626266
64343639396134313039336331376137646632633136333937376563666366636665343333613938
6162643231643131350a356463636662613830636237663066313331346330353330326138633632
61373063306662333733663433373430613266626466383632323363353536326163
[fedora-23:ansible (devel % u=)]$ cat include_var_ping.yml 
---
- hosts: localhost
  gather_facts: false
  tasks:
    - name: include an encrypted vars file with no .yml
      include_vars:
          file: vars_to_include.foo
          name: test_var

    - name: debug unsafe
      debug: var=test_var

[fedora-23:ansible (devel % u=)]$ ansible-playbook -vvv include_var_ping.yml --ask-vault-pass
Using /home/adrian/ansible/ansible.cfg as config file
Vault password: 
 [WARNING]: provided hosts list is empty, only localhost is available


PLAYBOOK: include_var_ping.yml *******************************************************************************************************************************
1 plays in include_var_ping.yml

PLAY [localhost] *********************************************************************************************************************************************

TASK [include an encrypted vars file with no .yml] ***********************************************************************************************************
task path: /home/adrian/src/ansible/include_var_ping.yml:5
ok: [localhost] => {
    "ansible_facts": {
        "test_var": {
            "baz": "foobar", 
            "blip": 37
        }
    }, 
    "changed": false, 
    "invocation": {
        "module_args": {
            "file": "vars_to_include.foo", 
            "name": "test_var"
        }, 
        "module_name": "include_vars"
    }
}

TASK [debug unsafe] ******************************************************************************************************************************************
task path: /home/adrian/src/ansible/include_var_ping.yml:10
ok: [localhost] => {
    "test_var": {
        "baz": "foobar", 
        "blip": 37
    }
}

PLAY RECAP ***************************************************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0  
```

**Comment by bcoca** [comment]
This was a bug when the dir option was added to include vars, fixed in #18324

**Comment by bcoca** [comment]
this is dupe of #18223
